File: jawa/boba-fett/src/main/java/ru/prohor/universe/bobafett/service/UnknownCommandHandler.java
package ru.prohor.universe.bobafett.service;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;
import org.telegram.telegrambots.meta.api.objects.Message;
import ru.prohor.universe.bobafett.command.Commands;
import ru.prohor.universe.jocasta.tgbots.api.FeedbackExecutor;
import ru.prohor.universe.jocasta.tgbots.api.UnknownActionKeyHandler;

@Service
public class UnknownCommandHandler implements UnknownActionKeyHandler<Message, String> {
    private static final Logger log = LoggerFactory.getLogger(UnknownCommandHandler.class);

    @Override
    public void handleUnknownActionKey(Message message, String command, FeedbackExecutor feedbackExecutor) {
        log.trace("unknown command '{}' from chat with id {}", command, message.getChatId());
        feedbackExecutor.sendMessage(
                message.getChatId(),
                "Неизвестная команда. Посмотреть список доступных команд - " + Commands.COMMANDS
        );
    }
}

File: jawa/yoda/yoda-bot/src/main/kotlin/setup/UnknownCommandHandler.kt
package ru.prohor.universe.yoda.bot.setup

import org.springframework.stereotype.Service
import org.telegram.telegrambots.meta.api.objects.Message
import ru.prohor.universe.jocasta.tgbots.api.FeedbackExecutor
import ru.prohor.universe.jocasta.tgbots.api.UnknownActionKeyHandler
import ru.prohor.universe.yoda.bot.Text.CommandReplies

@Service
class UnknownCommandHandler : UnknownActionKeyHandler<Message, String> {
    override fun handleUnknownActionKey(
        message: Message,
        key: String,
        feedbackExecutor: FeedbackExecutor
    ) {
        val chat = message.chat
        if (chat.isUserChat) {
            feedbackExecutor.sendMessage(chat.id, CommandReplies.unknown())
        }
    }
}
